### ISSUE TITLE
subgraphs don't inherit format

### ISSUE THREAD

**Issue by BMaxV** [xflr6/graphviz#114]
Hello again,

So I thought about how it might be useful to save the source or render subgraphs.

It appears that this code:

```
import graphviz

d=graphviz.Digraph()
d.format="svg"
d.attr(shape="box")
with d.subgraph(name="cluster_1") as sub:
    sub.node("1","hello")
    sub.node("2","world")
    sub.edge("1","2")
    sub.render(sub.name)
with d.subgraph(name="cluster_2") as sub:
    sub.node("3","hello2")
    sub.node("4","world2")
    sub.edge("3","4")
    sub.render(sub.name)
d.render("ex")
```

Produces a big svg and two smaller pdfs because that's the default.

I looked at the code a bit and it seems like the format is being inherited from base and the parent attribute isn't considered.

I could write a PR, but we didn't finish with the first one #113 yet.

**Comment by xflr6** [comment]
Thanks. I am not sure how good of a use case it is to `render()` a subgraph that was created via the context-manager (if a subgraph is meaningful also on its own, one might rather create the subgraphs independently and then add them to the (super)graph). In other words, this was not envisioned.

On the other hand, I think I agree that users probably expect the subgraph in the context manager to inherit attributes from the supergraph, candidates: `directory`, `format`, `engine`, `encoding` and `strict`.

However, IIUC none of them are used when rendering the supergraph, so the question of inheritance would only be relevant for serializing/rendering a subgraph created via context-manager.

**Comment by xflr6** [comment]
This will be a (slight) behavioural change so we should test and document it well.

Do you want to go ahead and create a PR to copy the five attributes from the parent at subgraph creation?

**Comment by xflr6** [comment]
Sorry for the delay: Fixed in [0.15 release](https://pypi.org/project/graphviz/0.15/#files) (885220d49eee1f04cd6493cc69acd72684c59e05).
